### ISSUE TITLE
empty request in middleware

### ISSUE THREAD

**Issue by fzhan** [laravel/framework#14317]
Version 5.2.39

in any middleware handle function trying to dd($request->all()), turns out empty.

Any tips?

**Comment by GrahamCampbell** [comment]
Please ask on the forums, and I'm sure someone can help you out there.

**Comment by fzhan** [comment]
Let me rephrase the question, @GrahamCampbell We run 5.2.38 on production site, using LaravelLocalization on its master branch latest version, nothing was changed until we `composer update` and received 5.2.39, its redirect middleware receives empty request.

I've also posted the issue here: https://github.com/mcamara/laravel-localization/issues/336

**Comment by GrahamCampbell** [comment]
Oh, right. Sorry about that! Ping @taylorotwell.

**Comment by fzhan** [comment]
This turns out to be a multiple issue:

1. after upgrading to 5.2.39, some how redirect won't remember the correct url.intend:
2. See here: [mcamara/laravel-localization](https://github.com/mcamara/laravel-localization/issues/336) when using localization on oauth/authorize, the request parameter was lost during the redirect from oauth/authorize to  /en/oauth/authorize.
3. /en/oauth/authorize is asking to check the auth middleware, without carrying the correct request (client_id etc.) redirect to login, causing 'url.intende' to record only '/en/oauth/authorize'

I'm not entirely sure what's causing the issue here. It seems the logic in middleware has changed after 5.2.39. I'm posting my findings in all three package.

**Comment by taylorotwell** [comment]
I'm not sure if any middlewares have changed in these releases?

**Comment by themsaid** [comment]
@fzhan tested these cases on `5.2.39`:

1. `$request->all()` in all `Middleware::handle()` returns an array of parameters as usual.
2. `url.intended` stores the correct URL and laravel redirects to it after a successfull log in as usual.

Could it be some package you use that causes such behaviour?

**Comment by fzhan** [comment]
@themsaid It could, as I said there's a chain of packages involved here, and we have solved the problem by manually setting request()->fullUrl() instead of url()->current() in other packages, I'm still testing for bugs, will commit a solution to this issue asap.
